### ISSUE TITLE
unable to import @joplin/turndown into my esm project

### ISSUE THREAD

**Issue by vassudanagunta** [laurent22/joplin#10640]
### Operating system

macOS

### Joplin version

latest

### Current behaviour

```
import TurndownService from '@joplin/turndown'
```

results in 

```
error TS7016: Could not find a declaration file for module '@joplin/turndown'. '.../node_modules/@joplin/turndown/lib/turndown.cjs.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/joplin__turndown` if it exists or add a new declaration (.d.ts) file containing `declare module '@joplin/turndown';`
```

The same thing happens for `@joplin/turndown-plugin-gfm`.

### Expected behaviour

Looking in the `node_modules/@joplin/turndown`, I only see `turndown.cjs.js`, not the `.es.mjs` version that the package.json build script mentions.

I was hoping to use your version of Turndown.js!

**Comment by personalizedrefrigerator** [comment]
Based on `packages/turndown/package.json`, it seems that only the CJS version of the library is built in CI:
https://github.com/laurent22/joplin/blob/8eda8d3c849c01b5f3c79662afaf004e434d61b1/packages/turndown/package.json#L44
https://github.com/laurent22/joplin/blob/8eda8d3c849c01b5f3c79662afaf004e434d61b1/packages/turndown/package.json#L50

The `prepare` step might need to be changed to `npm run build-all` to support ESM.

**Comment by vassudanagunta** [comment]
This is still a needed fix. I hesitate to take a stab at it as I don't (yet) use Joplin and I not sure I can do the update safely without breaking the primary client (Joplin).

**Comment by vassudanagunta** [comment]
Yes, this really ought to be fixed. @personalizedrefrigerator seems to know what's what. I was hoping it would happen. I'm not familiar at all with the codebase and build system, otherwise I'd make a PR.

**Comment by vassudanagunta** [comment]
should keep reviving this each month?

**Comment by florian-sattler** [comment]
I would also find this feature very useful. Could you please provide guidance on how to implement the fix? I'd be happy to submit a PR if they are accepted.

**Comment by florian-sattler** [comment]
Update: I managed to compile a `.esm.js` file by adding a manual build script - however the rollup version was very old and did not support `mjs` as a build target. Upgrading to a newer version allowed me to create this exact file, however it broke the rest of the builds.

**Comment by willemmulder** [comment]
Would also really like to have this fixed 'by default' so that just installing @joplin/turndown works without complaints.
